src/python/pants/backend/shell/target_types.py

```diff
@@ -410,10 +417,14 @@ class ShellCommandLogOutputField(BoolField):
     help = "Set to true if you want the output from the command logged to the console."
 
 
-class ShellCommandRunWorkdirField(StringField):
+class ShellCommandWorkdirField(StringField):
     alias = "workdir"
-    default = "."
-    help = "Sets the current working directory of the command, relative to the project root."
+    default = None
+    help = (
+        "Sets the current working directory of the command, relative to the project root. If not "
+        "set, use the project root. To use the location of the `BUILD` file, use "
+        '`f"{build_file_dir()}"`.'
```

Comment:
Will `"."` also work? It's probably what I'd reach for without checking any docs

Reply:
It makes all the tests time out, so I'd say "no" :)

Reply:
😬 oof

Reply:
IMO it would be nice to make `.` work before stabilization - I expect it's what most people will think to use first. My brain also immediately connected the new `workdir` semantics to how `context_root` works in the `docker` subsystem, but `.` works for docker and not for this... which might be ok, I just always wish for more consistency between the various subsystems 😄

Reply:
I definitely agree on the need for consistency!

Note that the documentation currently says _relative to the project root_, which is important for the use in `experimental_run_shell_command`: `run` runs in the project root, so making `.` mean anything else would be super confusing to reason about.

I'm not certain I like the idea of `.` meaning one thing for sandbox processes (i.e. the `BUILD`file location) and another for interactive processes (i.e. the project root), and I'm also not certain that `.` could realistically mean anything other than the project root for interactive processes.

Reply:
As the desire for using the `BUILD` file location (i.e. `build_file_dir()`) may be expected to be relatively common, it would be nice if it were easier than relying on calling the `build_file_dir` function (or potentially worse, having to spell out the full path). 

My reasoning for using `.` to mean the `BUILD` file location is that you may think of the `BUILD` location as the current working directory for the targets defined therein, and as such, `.` translating to `build_file_dir()` makes perfect sense. This is not only used for docker, but for the visibility rules as well now.

I agree that it would be most preferable to have `.` mean the same regardless of kind of process; I don't see why `.` would have to be any different for interactive processes..?

Reply:
@kaos it's mostly just that internally, `.` means "relative to the project root" for `InteractiveProcess`es (i.e. internally Pants itself), and currently `experimental_run_shell_command` just naïvely passes that value directly into `InteractiveProcess`.

I'm happy with `.` to mean `build_file_dir()` in this case, and perhaps `/` can indicate the build/project root? Technically the empty string would also work for the root, but having a Truthy string value feels nicer to me.

Something like the following?

```
"."            # location of `BUILD_FILE`
"/"            # build/project root
""             # _also_ build/project root
"./foo"        # equivalent to `f"{build_file_dir()}/foo"`
"/foo"         # the dir `foo` at the build/project root
"foo"          # the dir `foo` at the build/project root

```

Reply:
Perfect. This aligns with my view of the world (in pants) and so with the features I've implemented ;)

Reply:
@kaos @danxmoran Does the current status match your understanding of things? (see https://github.com/pantsbuild/pants/pull/17928/commits/6474d3ab5a1d6439e8adfd582bd64aaaa531fd22)